File: Extract_yyyy_mm_dd.py
s = input().strip()
if len(s) == 10 and s[4] == '-' and s[7] == '-' and s[:4].isdigit() and s[5:7].isdigit() and s[8:].isdigit():
    y,m,d = int(s[:4]), int(s[5:7]), int(s[8:])
    print(f"{y} {m} {d}" if 1 <= m <=12 and 1 <= d <= 31 else "INCORRECT")
else:
    print("INCORRECT")

File: Extract_yyyy_mm_dd.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    string s;
    cin >> s;
    if (s.size() != 10 || s[4] != '-' || s[7] != '-')
        return cout << "INCORRECT",0;
    for (int i : {0, 1, 2, 3, 5, 6, 8, 9})
        if (!isdigit(s[i])) return cout << "INCORRECT",0;
    int y = stoi(s.substr(0, 4)), m = stoi(s.substr(5,2)), d = stoi(s.substr(8,2));
    if (m < 1 || m > 12 || d < 1 || d > 31) cout<<"INCORRECT";
    else cout << y << " " << m << " " << d;
}
